app/index.js

```diff
@@ -1,86 +1,103 @@
       store: true,
       default: 'MIT',
-    }, {
+    },
+    {
       name: 'githubUsername',
       message: 'github username:',
       store: true,
       validate: function (val) {
-        return val.length > 0 ? true : 'You have to provide a username';
+        return val.length > 0 ? true : 'You have to provide a username'
       }
-    }, {
+    },
+    {
+      type: "confirm",
+      name: "changelog",
+      message: "changelog:",
+      store: true,
+      default: true
+    },
+    {
       name: 'website',
       message: 'website:',
       store: true,
-      validate: function (val) {
-        return val.length > 0 ? true : 'You have to provide a website URL';
+      default: function (props) {
+        return "http://github.com/" + props.githubUsername
       },
       filter: function (val) {
-        return normalizeUrl(val);
+        return normalizeUrl(val)
       }
-    }], function (props) {
-      var tpl = {
-        moduleName: props.moduleName,
-        moduleDesc: props.moduleDesc,
-        moduleKeywords: (props.moduleKeywords || '').trim().split(',').map(function(i) { return (i || '').trim(); }),
-        moduleVersion: props.moduleVersion,
-        moduleLicense: props.moduleLicense,
-        camelModuleName: _s.camelize(props.moduleName),
-        githubUsername: props.githubUsername,
-        name: this.user.git.name(),
-        email: this.user.git.email(),
-        website: props.website,
-        humanizedWebsite: humanizeUrl(props.website)
-      };
+    }],
+    function (props) {
+      this.moduleName = props.moduleName
+      this.moduleDesc = props.moduleDesc
+      this.moduleVersion = props.moduleVersion
+      this.moduleLicense = props.moduleLicense
+      this.camelModuleName = _s.camelize(props.moduleName)
+      this.githubUsername = props.githubUsername
+      this.name = this.user.git.name()
+      this.email = this.user.git.email()
+      this.website = props.website
+      this.humanizedWebsite = humanizeUrl(props.website)
+      this.moduleKeywords = (props.moduleKeywords || '')
+        .trim()
+        .split(',')
+        .map(function(i) {
+          return (i || '').trim()
+        })
+      this.hasChangelog = props.changelog
 
-      var cpTpl = function (from, to) {
-        this.fs.copyTpl(this.templatePath(from), this.destinationPath(to), tpl);
-      }.bind(this);
-
-      cpTpl('_index.js',     'index.js');
-      cpTpl('_package.json', 'package.json');
-      cpTpl('_README.md',     'README.md');
-      cpTpl('_test.js',      'test.js');
-      cpTpl('editorconfig',  '.editorconfig');
-      cpTpl('gitignore',     '.gitignore');
-      cpTpl('npmignore',     '.npmignore');
-      cpTpl('travis.yml',    '.travis.yml');
+      this.template('package.json')
+      this.template('README.md')
+      if (this.hasChangelog) {
+        this.template("CHANGELOG.md")
+      }
+      this.template('index.js',      'src/index.js')
+      this.template('test.js',       'test/index.js')
```

Comment:
if you dont mind i prefer to have module’s structure as flat as possible, so can you keep this thing in previous state